### PR TITLE
Feat/menu commands updating

### LINKED ISSUE #175: Support more (squeak) menu commands
As a user I want to find as many commands as possible in Algernon in order to not click through menus

## Acceptance Criteria
- [x] Make all top-level menu bar items accessible in Algernon
- [x] don't forget to test

### PR DESCRIPTION
**Pull request by niklasmohrin**
Closes #175

### CODE REVIEW

File: .github/workflows/smalltalk-ci.yml [hunk 1]
```diff
@@ -5,22 +5,20 @@ jobs:
...
+    - uses: actions/checkout@v2
+    - uses: hpi-swa/setup-smalltalkCI@v1
+      id: smalltalkci
+      with:
+        smalltalk-version: ${{ matrix.smalltalk }}
+    - run: smalltalkci -s ${{ steps.smalltalkci.outputs.smalltalk-version }}
+      shell: bash
+      timeout-minutes: 15
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (FranzSw) [inline .github/workflows/smalltalk-ci.yml]: This is an old version (`smalltalk-image` is deprecated), I guess the changes were accidental or Squeak does weird stuff again?

**Review by FranzSw — CHANGES_REQUESTED** [review]
Not sure why, but `.coverage.ston`, `coverage-reporter.yml` are deleted and the issue below.

**Review by FranzSw — APPROVED** [review]

**Review by bewee — APPROVED** [review]

File: packages/Algernon.package/ALGWorldMenuType.class/class/filterMenuItems..st [hunk 2]
```diff
@@ -0,0 +1,4 @@
+helpers
+filterMenuItems: aMorph
```
Comment (niklasmohrin) [inline packages/Algernon.package/ALGWorldMenuType.class/class/filterMenuItems..st]: hm, there isn't really any filtering of menu items going on, is there?

what about `scrapeMenuItems`? I fell like "filter" / "select" reads as if the `aMorph` was a collection, though it isn't (and the name `each` doesn't make that clear either)

File: packages/Algernon.package/ALGWorldMenuType.class/class/menuEntries.st [hunk 3]
```diff
@@ -0,0 +1,17 @@
+helpers
```
Comment (niklasmohrin) [inline packages/Algernon.package/ALGWorldMenuType.class/class/menuEntries.st]: Why "helpers" exactly? It seems to me that this is _strongly_ tied to the generation of the algtermlist and having the method in it's own category doesn't do code navigation any good either

File: packages/Algernon.package/ALGStaticWorldMenuType.class/class/filterMenuItems..st [hunk 4]
```diff
@@ -0,0 +1,8 @@
+helpers
+filterMenuItems: aMorph
+	| entries |
+	entries := OrderedCollection new.
+	(aMorph isKindOf: DockingBarUpdatingMenuMorph) 
+		ifFalse: [ entries addAll: aMorph submorphs ].
+						
+	^ entries
```
Comment (niklasmohrin) [inline packages/Algernon.package/ALGStaticWorldMenuType.class/class/filterMenuItems..st]:
```suggestion
helpers
filterMenuItems: aMorph
	^ (aMorph isKindOf: DockingBarUpdatingMenuMorph) 
		ifTrue: [ OrderedCollection new ]
		ifFalse: [ OrderedCollection newFrom: aMorph submorphs ].
```

**Review by niklasmohrin — COMMENTED** [review]
Sorry for the late review :sweat_smile:

File: packages/Algernon.package/ALGWorldMenuType.class/class/filterMenuItems..st (hunk 2 above)
Comment (Olliwehr) [reply packages/Algernon.package/ALGWorldMenuType.class/class/filterMenuItems..st]: Yeah, you're right. I would suggest sth like `extractMenuEntriesFromMenuMorph` (or `extractMenuEntriesFrom: aMenuMorph` for short) as we do exactly that, no?

File: packages/Algernon.package/ALGWorldMenuType.class/class/menuEntries.st (hunk 3 above)
Comment (Olliwehr) [reply packages/Algernon.package/ALGWorldMenuType.class/class/menuEntries.st]: This then also applies to the other methods in this class (`menus`, `filterMenuItems`, `dockingBar`), doesn't it?
So what about putting all those in the category `algertem lists`?

File: packages/Algernon.package/ALGStaticWorldMenuType.class/class/filterMenuItems..st (hunk 4 above)
Comment (Olliwehr) [reply packages/Algernon.package/ALGStaticWorldMenuType.class/class/filterMenuItems..st]: Quite a nice suggestion. The current implementation seems to be similar in structure to the same method in `ALGDynamicWorldMenuType`. Should we also adjust the implementation there or is that not so relevant?

File: packages/Algernon.package/ALGWorldMenuType.class/class/menuEntries.st (hunk 3 above)
Comment (bewee) [reply packages/Algernon.package/ALGWorldMenuType.class/class/menuEntries.st]: I had a look at the other types. Some put their algertem list related methods into the category `algertem list`, others put them into other categories like `helper` or `instance creation`. So I think putting them all into `algertem list` and removing the `helpers` category is fine.

File: packages/Algernon.package/ALGStaticWorldMenuType.class/class/filterMenuItems..st (hunk 4 above)
Comment (bewee) [reply packages/Algernon.package/ALGStaticWorldMenuType.class/class/filterMenuItems..st]: I think so

**Comment by bewee** [comment]
Why does this **always** happen when I push a commit?😬😓😤

File: packages/Algernon.package/ALGDynamicWorldMenuType.class/class/extractMenuEntriesFrom..st [hunk 5]
```diff
@@ -0,0 +1,9 @@
+algertem lists
+extractMenuEntriesFrom: aMorph
+	^ (aMorph isKindOf: DockingBarUpdatingMenuMorph) 
+		ifTrue: [ | entries |
+				aMorph updateMenu. 
+				entries := aMorph submorphs.
```
Comment (niklasmohrin) [inline packages/Algernon.package/ALGDynamicWorldMenuType.class/class/extractMenuEntriesFrom..st]: From `Morph>>submorphs`:

> This method returns my actual submorphs collection. Modifying the collection directly could be dangerous; make a copy if you need to alter it.

I guess it should be `OrderedCollection newFrom: aMorph submorphs` to be safe

**Review by niklasmohrin — COMMENTED** [review]
Sorry for nitpicking, I am gonna go ahead and commit this one myself :smile:

**Comment by LinqLover** [comment]
> Why does this **always** happen when I push a commit?😬😓😤

This is unfortunately a known bug in Squot: https://github.com/hpi-swa/Squot/issues/199 Usually it should help to commit everything before pulling (technically, this allows the pull to act as a fast-forward merge).

**Comment by bewee** [comment]
So can we now merge this?

**Pull request merged**